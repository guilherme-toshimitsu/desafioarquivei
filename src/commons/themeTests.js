import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
