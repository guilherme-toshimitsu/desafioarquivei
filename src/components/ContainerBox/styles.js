import { memo } from "react";
import styled from "styled-components";
import { breakpoints, points } from "@commons/theme";

const StyledContentBox = styled.div`
  box-sizing: border-box;
  position: relative;
  margin: 0 auto;
  min-height: calc(100vh - 92px - 60px);
  min-width: 320px;
  overflow: hidden;
  background-color: #F0F0F0;

  ${breakpoints.sm} {
    padding: 8px,
    maxWidth: ${points.sm}
    margin: 0px 0px 0px 10px;
  }

  ${breakpoints.md} {
    padding: 8px,
    maxWidth: ${points.md}
  }

  ${breakpoints.lg} {
    padding: 8px,
    maxWidth: ${points.lg}
  }
`;

export default memo(StyledContentBox);
