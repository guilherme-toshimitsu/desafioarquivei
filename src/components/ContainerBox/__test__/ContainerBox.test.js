import React from "react";
import ContainerBox from "../";
import { renderWithTheme } from "@commons/themeTests";

describe("<ContainerBox />", () => {
  it("creates a ContainerBox component", () => {
    const wrapper = renderWithTheme(<ContainerBox />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
