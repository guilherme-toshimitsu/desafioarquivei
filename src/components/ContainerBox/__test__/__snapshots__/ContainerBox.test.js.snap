// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerBox /> creates a ContainerBox component 1`] = `
<DocumentFragment>
  <div
    class="ContainerBox__StyledContentBox-sc-18d78nw-0 sYzDb"
  />
</DocumentFragment>
`;
